IFTB.bs

```diff
@@ -0,0 +1,535 @@
+
+     Load each needed chunk using individual files:
+     1.  Calculate the hex string for the chunk index
+     2.  Get the chunkFilesURI string from the IFTB table and substitute "$1", "$2", ... with
+          the corresponding hex digit.
+     3.  Combine the relative URI with the URL of the original font to get the URL for the chunk.
+     4.  Request that file via HTTP.
+
+     Load each needed chunk from the range file:
+     1.  Get the chunkRangeFileURI from the string IFTB table.
+     2.  Combine that URI with the URL of the original font to get the URL for the range file.
+     3.  Get the byte range for each chunk from the chunkOffsetTable in the IFTB table.
+     4.  Form a HTTP range request for those byte ranges using the combined URL.
+8.   Decompress each retrieved chunk content.
+9.   Merge the glyph contents in all of the chunks into the corresponding tables, reallocating the
+     memory for the font if necessary. (All of the relevant tables use an offset array and concatenated
+     glyph data convention.)
+10.  Update the IFTB table to record the chunks added (the table will not change in length).
+11.  Update the "sfnt" headers for the correct offsets, sizes, and (if needed by the client) checksums.
+     (Only the trailing CFF, CFF2, glyf, loca and or gvar tables will change in size or location.) Also
+     update the checksumAdjustment in the head table (if needed by the client).
+12.  If caching, store the file at this point.
+13.  Update the SFNT version to 0x00010000 (for a glyf table) or "OTTO" (for a CFF or CFF2 table) and
+     also the head table checksumAdjustment again (if needed by the client).
+14. Render the text with the updated font.
+
+### Subsequent Requests ### {#iftb-browser-behaviors-subsequent-requests}
+
+Subsequent requests proceed like the initial request starting at step 3.
+
+IFTB Encoding {#iftb-encoding}
+------------------------------
+
+An input font is encoded in the IFTB format using an encoder as a separate, initial step.  If any other
+subsetting (such as removing or reducing variable axes, or removing glyphs that are not needed) is
+desired the input font should be subset prior to IFTB encoding with an appropriate tool.
+
+With IFTB only glyph data is transferred incrementally. The encoded file also
+"retains" GIDS the input font's GIDs, and therefore most tables can simply be
+copied from the source font.  However, an encoder can also deserialize and
+reserialize tables if it desires.
+
+The details of the encoding process may differ by encoder and are beyond the scope of this document.
+However, all mappings of codepoints (though GIDs) to bins and bins to feature-specific bins must meet
+this closure requirement:
+
+* The set of glyphs contained in the chunks loaded through the GID and feature maps must be a superset
+    of those in the GID closure of the font subset description.
+
+The encoder has three options for addressing with joint dependencies on individual glyphs:
```

Comment:
I would clearly mark this part (the 3 options below) as being non-normative since it's just providing advice on how an encoder might be built and not actually laying out requirements that an encoder must follow.

Reply:
Or another option this could be re-framed as things that the encoder may do which are valid. For example it's valid to place a glyph in more than one chunk if needed.